resources/Microsoft.Windows.Setting.Accessibility/Microsoft.Windows.Setting.Accessibility.psm1

```diff
@@ -419,6 +421,179 @@ class Audio
+    {
+		$TextCursorIndicatorColorArguments = @{  Path = $global:CursorIndicatorAccessibilityRegistryPath; Name = ([TextCursor]::TextCursorIndicatorColorProperty)}
+        if (-not(DoesRegistryKeyPropertyExist @TextCursorIndicatorColorArguments))
+        {
+            return $false
+        }
+        else
+        {
+            $textCursorSetting = (Get-ItemProperty @TextCursorIndicatorColorArguments).IndicatorColor
+            return $textCursorSetting
+        }        
+    }
+
+    static [int] GetTextCursorThickness()
+    {
+		$TextCursorThicknessArguments = @{ Path = $global:ControlPanelDesktopRegistryPath; Name = ([TextCursor]::TextCursorThicknessProperty); }
+        if (-not(DoesRegistryKeyPropertyExist @TextCursorThicknessArguments))
+        {
+            return 1
+        }
+        else
+        {
+            $textCursorSetting = (Get-ItemProperty @TextCursorThicknessArguments).CaretWidth
+            return $textCursorSetting
+        }        
+    }
+
+    [TextCursor] Get()
+    {
+        $currentState = [TextCursor]::new()
+        $currentState.TextCursorIndicatorStatus = [TextCursor]::GetTextCursorIndicatorStatus()
+        $currentState.TextCursorIndicatorSize = [TextCursor]::GetTextCursorIndicatorSize()
+        $currentState.TextCursorIndicatorColor = [TextCursor]::GetTextCursorIndicatorColor()
+        $currentState.TextCursorThickness = [TextCursor]::GetTextCursorThickness()
+        
+        return $currentState
+    }
+
+    [bool] Test()
+    {
+        $currentState = $this.Get()
+        if (($null -ne $this.TextCursorIndicatorStatus) -and ($this.TextCursorIndicatorStatus -ne $currentState.TextCursorIndicatorStatus))
+        {
+            return $false
+        }
+        if ((0 -ne $this.TextCursorIndicatorSize) -and ($this.TextCursorIndicatorSize -ne $currentState.TextCursorIndicatorSize))
+        {
+            return $false
+        }
+        if ((0 -ne $this.TextCursorIndicatorColor) -and ($this.TextCursorIndicatorColor -ne $currentState.TextCursorIndicatorColor))
+        {
+            return $false
+        }
+        if ((0 -ne $this.TextCursorThickness) -and ($this.TextCursorThickness -ne $currentState.TextCursorThickness))
+        {
+            return $false
+        }
+
+        return $true
+    }
+
+    [void] Set()
+    {
+        if (-not $this.Test())
+        {
+			$TextCursorIndicatorStatusArguments = @{ Path = $global:NTAccessibilityRegistryPath; Name = ([TextCursor]::TextCursorIndicatorStatusProperty); }
+			$TextCursorThicknessArguments = @{ Path = $global:ControlPanelDesktopRegistryPath; Name = ([TextCursor]::TextCursorThicknessProperty); }
+			$TextCursorIndicatorSizeArguments = @{  Path = $global:CursorIndicatorAccessibilityRegistryPath; Name = ([TextCursor]::TextCursorIndicatorSizeProperty)}
+			$TextCursorIndicatorColorArguments = @{  Path = $global:CursorIndicatorAccessibilityRegistryPath; Name = ([TextCursor]::TextCursorIndicatorColorProperty)}

```

Comment:
Move these inside their respective if statements. That way if the condition is not satisfied, these arguments won't be created.